app/models/user.rb

```diff
@@ -0,0 +1,32 @@
+class User < ApplicationRecord
+  has_many :created_suggestions, class_name: :Suggestion, foreign_key: :author_id
+  has_many :thumbups , class_name: :ThumbsUp, foreign_key: :voter_id
+  has_many :voted_suggestions, through: :thumbups, source: :suggestion
+  has_many :votes , through: :created_suggestions
+  has_many :supported_suggestions, through: :votes, source: :voter
+  has_many :supports_suggestions , through: :voted_suggestions, source: :author
```

Comment:
These associations need to be thought through some more. e.g. supported_suggestions is currently returning  a collection of users